internal/config/authentication.go

```diff
@@ -441,10 +441,7 @@ func (a AuthenticationMethodOIDCConfig) info() AuthenticationMethodInfo {
 	// this ensures we expose the authorize and callback URL endpoint
 	// to the UI via the /auth/v1/method endpoint
 	for provider := range a.Providers {
-		providers[provider] = map[string]any{
-			"authorize_url": fmt.Sprintf("/auth/v1/method/oidc/%s/authorize", provider),
-			"callback_url":  fmt.Sprintf("/auth/v1/method/oidc/%s/callback", provider),
-		}
+		providers[provider] = map[string]any{}
```

Comment:
I wonder if there is a way we can preserve this. So as not to remove something from the API that by chance someone might depend upon.

We could always try and manipulate these provider URLs here when rendering the config before marshalling to JSON:
https://github.com/flipt-io/flipt/blob/main/internal/server/metadata/server.go#L38

Reply:
It looks there another place for manipulation here

https://github.com/flipt-io/flipt/blob/07208a6e346df8313597f45e73557ef495004704/internal/server/authn/public/server.go#L40-L44

Reply:
I reverted some part of my code and returned metadata for auth methods. Also I gave a try to embed the `prefix` into `callback_url` and `authorize_url` and I don't think it makes no benefits right now. Here is an example for `github` provider 

```go
func (s *Server) ListAuthenticationMethods(ctx context.Context, _ *emptypb.Empty) (*auth.ListAuthenticationMethodsResponse, error) {
	if md, ok := metadata.FromIncomingContext(ctx); ok {
		if forwardPrefix := md.Get("x-forwarded-prefix"); len(forwardPrefix) > 0 {
			prefix := path.Join(forwardPrefix...)
			resp := &auth.ListAuthenticationMethodsResponse{}
			for _, method := range s.resp.Methods {
				switch method.Method {
				case auth.Method_METHOD_GITHUB:
					metadata := method.Metadata
					if metadata != nil && metadata.Fields["authorize_url"] != nil {
						metadata, _ = structpb.NewStruct(metadata.AsMap())
						metadata.Fields["authorize_url"] = structpb.NewStringValue(path.Join(prefix, metadata.Fields["authorize_url"].GetStringValue()))
					}
					resp.Methods = append(resp.Methods, &auth.MethodInfo{
						Method:            method.Method,
						Enabled:           method.Enabled,
						SessionCompatible: method.SessionCompatible,
						Metadata:          metadata,
					})
				}
			}
			return resp, nil
		}
	}
	return s.resp, nil
}
```

With `oidc`, the code will become more convoluted as all providers are encoded in the generic metadata field.

One option is to leave those URLs as they are, assuming teams that use them already prepend the necessary well-known url.

Another option is to add redirect_address as @tvcsantos proposes in his PR. However, this could disrupt authentication for some teams. Adding the prefix with my code would also cause issues.

WDYT about this @GeorgeMac? I would appreciate any suggestion to move forward.

Reply:
i'll admit I'm a bit out of the loop on this PR, but if I understand correctly, I think I agree that teams should add the prefix themselves to the `redirect_url` and not expect us to add the prefix

Reply:
This might sounds like madness, but what about threading this all the way down to the `AuthenticationMethodInfo.info` call by way of `context.Context`?

https://github.com/flipt-io/flipt/blob/004d6d9ae329baa22823983317bf0b4403b2a779/internal/config/authentication.go#L253-L263

To avoid making a dependency on gRPC in the config code, we could make a context key and function for setting the prefix in the config package itself. Then where you have your code right now in ListAuthenticationMethods, we would just pull the grpc metadata off and set it on the appropriate key for the config.
Then make .AllMethods() be called explicitly on each call to ListAuthenticationMethods?

That should remove all the gnarly protobuf struct stuff.
I could do a little spike on this if you want?

Reply:
I took a stab at it here https://github.com/flipt-io/flipt/commit/9c3fb630

See what you think. One benefit to this approach is we don't need a switch case in the ListAuthenticationMethods. So the behaviour is all put on the config type itself to do what is appropriate. I added a test around ListAuthenticationMethods service to try with and without the metadata prefix.

Reply:
oOo i like that!

Reply:
> I took a stab at it here [9c3fb630](https://github.com/flipt-io/flipt/commit/9c3fb630)
> 
> See what you think. One benefit to this approach is we don't need a switch case in the ListAuthenticationMethods. So the behaviour is all put on the config type itself to do what is appropriate. I added a test around ListAuthenticationMethods service to try with and without the metadata prefix.

@GeorgeMac thank you for the idea. I merged your changes to PR and will wire all other places